### ISSUE TITLE
E1017 with nested !Select

### ISSUE THREAD

**Issue by mikeshimkus** [aws-cloudformation/cfn-python-lint#1919]
*cfn-lint version: (`cfn-lint --version`)*
0.46.0

*Description of issue.*
When linting this CFT YAML:
```yaml
subnet0A:
    DependsOn: ipv6CidrBlock
    Type: 'AWS::EC2::Subnet'
    Properties:
      VpcId: !Ref vpc
      CidrBlock: !Select
        - !Select
          - 1
          - !Split
            - ','
            - !FindInMap
              - subnetMap
              - !Ref numAzs
              - !Ref numSubnets
        - 'Fn::Cidr':
            - !GetAtt
              - vpc
              - CidrBlock
            - !Select
              - 0
              - !Split
                - ','
                - !FindInMap
                  - subnetMap
                  - !Ref numAzs
                  - !Ref numSubnets
            - !FindInMap
              - subnetMap
              - maskTocidrBits
              - !Ref subnetMask
```

I get this error:

E1017 Select index should be an Integer or a function Ref or FindInMap for Resources/subnet0A/Properties/CidrBlock/Fn::Select

Template works fine.
thanks

Cfn-lint uses the [CloudFormation Resource Specifications](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) as the base to do validation. These files are included as part of the application version. Please update to the latest version of `cfn-lint` or update the spec files manually (`cfn-lint -u`)

**Comment by kddejong** [comment]
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-select.html#w2ab1c33c28c51c15 says it only supports `Ref` and `FindInMap`.  It looks like that may be inaccurate and you can also embed another select?

**Comment by mikeshimkus** [comment]
Correct, it seems to work for me anyway.
